src/include/miopen/mlir_miopen_wrapper.hpp

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2018 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef MIOPEN_GUARD_MLIR_MIOPEN_WRAPPER_HPP
+#define MIOPEN_GUARD_MLIR_MIOPEN_WRAPPER_HPP
+
+namespace mlir {
+typedef void* MlirHandle;
```

Comment:
Are you suggesting something like below in the header file?
`#define MLIRMIOPEN_VERSION_MAJOR 0`
`#define MLIRMIOPEN_VERSION_MINOR 1`

How does MIOpen take advantage of those? Also, I'm not sure about what problem can it be if MIOpen use an older version of libMLIRMIOpen.a? To clarify, we don't plan on to make it a public API yet. So if there an API change, we can always bump the commit id if necessary.

Reply:
Removed `mlir_miopen_wrapper.hpp` and have it exported from library instead.

Reply:
> How does MIOpen take advantage of those? 

This would allow the library to adapt to different versions of MLIRMIOpen.

> So if there an API change, we can always bump the commit id if necessary.

Imagine that we see regression and would like to test the same MIOpen with two different versions of MLIRMIOpen, just for triaging purposes. This is only one scenario among many possible cases.

> `#define MLIRMIOPEN_VERSION_MAJOR 0`
> `#define MLIRMIOPEN_VERSION_MINOR 1`

I think that a single counter, like `#define MLIRMIOPEN_VERSION_FLAT 0` is quite enough (actually, better than x.y. scheme due to simplicity).

Reply:
Gotcha, I will do exactly this `#define MLIRMIOPEN_VERSION_FLAT 0` in MLIR. When this review landed, it will end with 0. Then when every time there is a change in API, I will increment the counter.